datastore/miitopia/types/mii_tube_search_result.go

```diff
@@ -0,0 +1,136 @@
+
+	content := contentWritable.Bytes()
+
+	mtsr.WriteHeaderTo(writable, uint32(len(content)))
+
+	writable.Write(content)
+}
+
+// ExtractFrom extracts the MiiTubeSearchResult from the given readable
+func (mtsr *MiiTubeSearchResult) ExtractFrom(readable types.Readable) error {
+	var err error
+
+	err = mtsr.ExtractHeaderFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract MiiTubeSearchResult header. %s", err.Error())
+	}
+
+	err = mtsr.Result.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract MiiTubeSearchResult.Result. %s", err.Error())
+	}
+
+	err = mtsr.Count.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract MiiTubeSearchResult.Count. %s", err.Error())
+	}
+
+	err = mtsr.Page.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract MiiTubeSearchResult.Page. %s", err.Error())
+	}
+
+	err = mtsr.HasNext.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to extract MiiTubeSearchResult.ExtractNext. %s", err.Error())
+	}
+
+	return nil
+}
+
+// Copy returns a new copied instance of MiiTubeSearchResult
+func (mtsr *MiiTubeSearchResult) Copy() types.RVType {
+	copied := NewMiiTubeSearchResult()
+
+	copied.Result = mtsr.Result
+	copied.Count = mtsr.Count
+	copied.Page = mtsr.Page
+	copied.HasNext = mtsr.HasNext
+
+	return copied
+}
+
+// Equals checks if the given MiiTubeSearchResult contains the same data as the current MiiTubeSearchResult
+func (mtsr *MiiTubeSearchResult) Equals(o types.RVType) bool {
+	if _, ok := o.(*MiiTubeSearchResult); !ok {
+		return false
+	}
+
+	other := o.(*MiiTubeSearchResult)
+
+	if !mtsr.Result.Equals(other.Result) {
+		return false
+	}
+
+	if !mtsr.Count.Equals(other.Count) {
+		return false
+	}
+
+	if !mtsr.Page.Equals(other.Page) {
+		return false
+	}
+
+	return mtsr.HasNext.Equals(other.HasNext)
+}
+
+// String returns the string representation of the MiiTubeSearchResult
+func (mtsr *MiiTubeSearchResult) String() string {
+	return mtsr.FormatToString(0)
+}
+
+// FormatToString pretty-prints the MiiTubeSearchResult using the provided indentation level
+func (mtsr *MiiTubeSearchResult) FormatToString(indentationLevel int) string {
+	indentationValues := strings.Repeat("\t", indentationLevel+1)
+	indentationEnd := strings.Repeat("\t", indentationLevel)
+
+	var b strings.Builder
+
+	b.WriteString("MiiTubeSearchResult{\n")
+	b.WriteString(fmt.Sprintf("%sResult: %s,\n", indentationValues, mtsr.Result))
+	b.WriteString(fmt.Sprintf("%sCount: %s,\n", indentationValues, mtsr.Count))
+	b.WriteString(fmt.Sprintf("%sPage: %s,\n", indentationValues, mtsr.Page))
+	b.WriteString(fmt.Sprintf("%sHasNext: %s,\n", indentationValues, mtsr.HasNext))
+	b.WriteString(fmt.Sprintf("%s}", indentationEnd))
+
+	return b.String()
+}
+
+// NewMiiTubeSearchResult returns a new MiiTubeSearchResult
+func NewMiiTubeSearchResult() *MiiTubeSearchResult {
+	mtsr := &MiiTubeSearchResult{
+		Result:  types.NewList[*MiiTubeMiiInfo](),
```

Comment:
The `Result` list type must be initialized. See this example: https://github.com/PretendoNetwork/nex-protocols-go/blob/186b30adafa28f3b04134079caa82c949edfef2e/datastore/super-mario-maker/types/datastore_get_meta_by_owner_id_param.go#L132-L140